src/main/java/gyro/google/kms/CryptoKeyResource.java

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.kms;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.cloud.kms.v1.CryptoKey;
+import com.google.cloud.kms.v1.CryptoKey.CryptoKeyPurpose;
+import com.google.cloud.kms.v1.CryptoKeyPathName;
+import com.google.cloud.kms.v1.KeyManagementServiceClient;
+import com.google.cloud.kms.v1.KeyRingName;
+import com.google.protobuf.Duration;
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.Timestamp;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Min;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidationError;
+import gyro.google.Copyable;
+import gyro.google.GoogleResource;
+import gyro.google.util.Utils;
+
+/**
+ * Add a cryto key to a king ring.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    google::crypto-key crypto-key-example
+ *        key-ring: $(google::key-ring example-key-ring)
+ *        name: "crypto-key-ring-example"
+ *        purpose: ENCRYPT_DECRYPT
+ *        rotation-period: 1
+ *        next-rotation-date: "09/21/2020"
+ *        primary-key-version-id: 1
+ *
+ *        crypto-key-version-template
+ *            algorithm: GOOGLE_SYMMETRIC_ENCRYPTION
+ *            protection-level: SOFTWARE
+ *        end
+ *    end
+ */
+@Type("crypto-key")
+public class CryptoKeyResource extends GoogleResource implements Copyable<CryptoKey> {
+
+    private KeyRingResource keyRing;
+    private String name;
+    private Long rotationPeriod;
+    private CryptoKeyPurpose purpose;
+    private String nextRotationDate;
+    private CryptoKeyVersionTemplate cryptoKeyVersionTemplate;
+    private Map<String, String> labels;
+    private String primaryKeyVersionId;
```

Comment:
I feel this should be `CryptoKeyVersionResource` as we have that implemented,
also from the console the resource id you get is the full id and not just the number as you are expecting